synapse/handlers/set_password.py

```diff
@@ -32,14 +34,17 @@ def __init__(self, hs):
         self._device_handler = hs.get_device_handler()
 
     @defer.inlineCallbacks
-    def set_password(self, user_id, newpassword, requester=None):
+    def set_password(
+        self,
+        user_id: str,
+        new_password: str,
+        logout_devices: bool,
```

Comment:
Should this be `True` by default?

Reply:
I do not believe so. Every caller of this API currently provides a value for `logout_devices` so adding a default here didn't seem useful (and could be potentially confusing). I can add it if that fits the standard way Synapse handles these sort of things though.

Reply:
Hm, considering the function did it before by default it seems to make sure to have this set to `True` by default.

That's just my general thoughts though, rather than stemming from what is typically done in Synapse. I can't think of any other examples at the moment.

Reply:
We had a brief discussions about this in #synapse-dev and it seems clearer to have callers be explicit and not having an unused default value is clearer.

I'm going to leave this as is!